x-pack/plugins/security_solution/public/common/components/event_details/event_fields_browser.tsx

```diff
@@ -203,23 +218,42 @@ export const EventFieldsBrowser = React.memo<Props>(
     const toggleColumn = useCallback(
       (column: ColumnHeaderOptions) => {
         if (columnHeaders.some((c) => c.id === column.id)) {
-          dispatch(
-            timelineActions.removeColumn({
-              columnId: column.id,
-              id: timelineId,
-            })
-          );
+          if (isTimelineScope(scopeId)) {
+            dispatch(
+              timelineActions.removeColumn({
+                columnId: column.id,
+                id: scopeId,
+              })
+            );
+          } else if (isInTableScope(scopeId)) {
+            dispatch(
+              dataTableActions.removeColumn({
+                columnId: column.id,
+                id: scopeId,
+              })
+            );
+          }
```

Comment:
NIT: Maybe we could have a helper function outside the component:
```
const getScopedActions = (scopeId: string) => {
      if (isTimelineScope(scopeId)) {
        return timelineActions;
      } else if (isInTableScope(scopeId)) {
        return dataTableActions;
      }
};
```
So we won't have to chain all these conditionals in the component callbacks.